todoapp-main/src/test/java/ru/todoapp/service/TaskServiceTest.java

```diff
@@ -0,0 +1,264 @@
+package ru.todoapp.service;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.kafka.core.KafkaTemplate;
+import ru.todoapp.model.GetTaskRequestEntity;
+import ru.todoapp.model.TaskEntity;
+import ru.todoapp.model.dto.*;
+import ru.todoapp.repository.RequestRepository;
+import ru.todoapp.repository.TaskRepository;
+import ru.todoapp.repository.UserRepository;
+import ru.todoapp.utils.KafkaTopics;
+
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class TaskServiceTest {
+    private AddTaskRequestDTO addTaskRequestDTO;
+
+    private RequestResultDTO successfulResult;
+
+    private RequestResultDTO failedResult;
+
+    @Mock
+    private KafkaTemplate<String, RequestResultDTO> requestResultDTOKafkaTemplateMock;
+
+    @Mock
+    private KafkaTemplate<String, FetchTasksResponseDTO> fetchTasksResponseDTOKafkaTemplateMock;
+
+    @Mock
+    private UserRepository userRepositoryMock;
+
+    @Mock
+    private RequestRepository requestRepositoryMock;
+
+    @Mock
+    private TaskRepository taskRepositoryMock;
+
+    @InjectMocks
+    private TaskService taskService;
+
+    @BeforeEach
+    void setUp() {
+        addTaskRequestDTO = new AddTaskRequestDTO("1", "2", "Do something", "2032-12-26T09:40:23+03:00");
+        successfulResult = RequestResultDTO.builder().requestUUID("1").status(RequestStatus.SUCCESS).build();
+        taskService = new TaskService(requestResultDTOKafkaTemplateMock,
+                fetchTasksResponseDTOKafkaTemplateMock,
+                requestRepositoryMock,
+                userRepositoryMock,
+                taskRepositoryMock);
+    }
+
+    @AfterEach
+    void tearDown() {
+    }
+
+    /**
+     * Test for successful completion of the method
+     */
+    @Test
+    void handleAdditionSuccessTest() {
+        when(userRepositoryMock.exists(any(String.class))).thenReturn(true);
```

Comment:
Good practise to comment given/when/then between 3 logical steps of the test.

Reply:
Ok